### ISSUE TITLE
Improve the documentation of PEP 495 features without referencing the PEP

### ISSUE THREAD

**Issue by abalkin** [python/cpython#101235]
Well, ideally we want the docs to avoid having to reference PEPs to explain things as far as the use of the current implementation goes, as opposed to the historical design decisions that shaped the eventual implementation in CPython. 

That said, if you really want, it's the https://github.com/python/cpython repo along with the rest of the source code. You can find a link to the source code of every docs page at the top of said page:

![image](https://user-images.githubusercontent.com/17051931/212808308-8cf895ed-9190-4670-bfaa-feb30423ae37.png)

The link for this particular page is https://github.com/python/cpython/tree/main/Lib/datetime.py

_Originally posted by @CAM-Gerlach in https://github.com/python/peps/issues/2959#issuecomment-1384810130_

**Comment by CAM-Gerlach** [comment]
Thanks, @abalkin . Just to clarify, I didn't mean to imply that the CPython docs should _never_ reference PEPs, but rather just that they shouldn't _rely_ on PEPs to "explain things", i.e. for the often-neglected [Explanation](https://diataxis.fr/explanation/) type content critical to reader understanding of the key concepts behind the implementation, and that parts of the PEP content should be ported over to the canonical docs, if likely to be useful to a sufficient cross-section of future users of a feature to teach, describe, instruct and explain the current implementation (as opposed to being more of historical/background interest, or more relevant to the decision-making process).

----

Also, porting over and summarizing some discussion relevant to this issue from python/peps#2959 :

@bxparks reported:

> ...the behavior of the datetime library (or any timezone library in any language) around the Gap is non-trivial, and it took me a long time (1-2 years) to understand all the nuances fully.

I replied:

> If text in the PEP was important to your understanding of the behavior of the datetime module, then perhaps it would be worth porting the relevant missing portions of the PEP's explanation to the Python docs, as an explanation-type section. Currently the docs cover the (API) "reference" topic thoroughly, but are somewhat more sparse in serving the other areas, particularly explanation. As the goal of explanation-type docs content is to serve understanding of the thing documented, this would seem like a good fit. if you're interested in something like that, I could help guide you through it (especially since I'm also on the Python Docs team).

@bxparks replied:

> My opinion is that an average software developer would find it very difficult to understand the `fold` parameter in the datetime library without reading PEP 495, with the correct Gap diagram. During an overlap (e.g. 02:00->01:00 shift in the fall-back), the `fold` can be hand-wave explained, as currently done in the `datetime.rst` page. But during a gap (e.g. 02:00->03:00 spring-forward), an accurate explanation of `fold` in the datetime docs page would need PEP 495.
> 
> An extreme example of such a gap is something like the `Pacific/Apia` time zone in Samoa, which has a 24 hour gap in Dec 2011, going from 2011-12-29 23:59:59 to 2011-12-31 00:00:00, skipping Dec 30 entirely. I think it would be very difficult to explain how the `fold` parameter handles this case without PEP 495.

and I clarified:

> It's certainly not any sort of hard rule; PEPs are referenced a number of places to explain the motivation, rationale and history of a specific feature, and it's better to reference the PEP than to simply leave the reader hanging completely when it comes to explanation-type content serving their understanding.
> 
> What I was suggesting, rather, is that while adding references to the PEP (`` :pep:`495` ``) is okay as a short term stopgap, what would be really helpful is if you could help migrate the relevant existing portions of the PEP that are important to reader understanding of the current `fold` behavior (as opposed to, e.g., justifying the changes, outlining backward compatibility, outlining alternatives, or specifying the API implementation) to a new top-level explanation section in the `datetime` docs on the topic.

**Comment by hugovk** [comment]
Is there something concrete to do here, or can we close this issue?

**Comment by StanFromIreland** [comment]
@picnixz I agree with you, this should be closed. Seeing there were no objections in ~3.5 months, I would say its time :-)

**Comment by picnixz** [comment]
I'm closing it as `not planned` but each aforementioned point can have its own issue if someone thinks they need to be fixed (the meaning of 'fold' seems to be lacking some precision but I didn't check nor do I know whether it's correct; strictly speaking, it seems that PEP-495 could be sufficient for now even if it's not perfect).
